master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -894,6 +896,14 @@ private[celeborn] class Master(
     val numReducers = requestSlots.partitionIdList.size()
     val shuffleKey = Utils.makeShuffleKey(requestSlots.applicationId, requestSlots.shuffleId)
 
+    if (numReducers > slotAssignMaxPartitions) {
+      logError(s"reject $shuffleKey to request $numReducers partitions, which exceeds $slotAssignMaxPartitions")
+      context.reply(RequestSlotsResponse(StatusCode.REQUEST_FAILED, new WorkerResource()))
+      return
+    } else if (numReducers > slotAssignWarningPartitions) {
+      logWarning(s"too much partitions, $shuffleKey requests $numReducers exceeds $slotAssignWarningPartitions")
```

Comment:
we audit the partitions, it is not necessary to log warning.
https://github.com/apache/celeborn/pull/3265